Supports hal_nxp of MCUX SDK 25.06.00

ZhaoxiangJin opened the pull request:
Update hal nxp to mcux sdk 25.06.00.

ZhaoxiangJin committed:
west.yml: Update hal_nxp to mcux sdk 25.06.00

Update hal_nxp to mcux sdk 25.06.00.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
modules: nxp: Add more MCUX SDK NG cmake variables

These variables are used to make MCUX SDK NG cmake
can find the right cmake files.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
drivers: interrupt_controller: Update CMakeLists to fix build error

Now use the 5-parameter function "PINT_PinInterruptConfig"
deprecated in MCUX SDK, need to add compile definition
'PINT_USE_LEGACY_CALBACK' to make intc_nxp_pint compatible
with updated 'fsl_pint' driver.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
modules: hal_nxp: Add core_id for RT685 HIFI4 core

1. Add core_id for RT685 HIFI4 core, this variable is
used in MCUX SDK cmake.
2. Set cmake variables 'CONFIG_MCUX_COMPONENT_driver.smartdma_mcxn'
and 'CONFIG_MCUX_COMPONENT_driver.smartdma_rt500' to "ON" as these
variables control the loading of source files in the latest driver
CMakeList.txt.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
scripts: compliance: Add items to UNDEF_KCONFIG_ALLOWLIST

'MCUX_HW_CORE' and 'MCUX_HW_FPU_TYPE' are used in
modules/hal_nxp/device.cmake. It is a variable used
by MCUX SDK CMake.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
boards: nxp: Renamed MCXA166 to MCXA346

1. Renamed MCXA166 to MCXA346.
2. NXP frdm_mcxa166 is renamed to frdm_mcxa346,
add this information to migration-guide-4.3.rst.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

peterwangsz committed:
boards: nxp: Renamed MCXA276 to MCXA266

1. Renamed MCXA276 to MCXA266
2. NXP frdm_mcxa276 is renamed to frdm_mcxa266,
add this information to migration-guide-4.3.rst.

Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

kartben commented on doc/releases/release-notes-4.2.rst:
```diff
@@ -627,8 +627,8 @@ New Boards
    * :zephyr:board:`frdm_mcxa153` (``frdm_mcxa153``)
-   * :zephyr:board:`frdm_mcxa166` (``frdm_mcxa166``)
-   * :zephyr:board:`frdm_mcxa276` (``frdm_mcxa276``)
+   * :zephyr:board:`frdm_mcxa346` (``frdm_mcxa346``)
+   * :zephyr:board:`frdm_mcxa266` (``frdm_mcxa266``)
```
This should not be renamed, board wasn't named like this at the time of the release. Just drop the `zephyr:board:` role I guess

ZhaoxiangJin replied:
I also think this files should not be changed, but the CI will check the 'release-notes-4.2.rst' and report board not find issue.

ZhaoxiangJin replied:
I dropped the change, the CI show failures on documentation build, I'd like to know why CI still builds release-notes-4.2.rst even though 4.2 has been released.

dleach02 replied:
Good question. 

@kartben this seems like a new case for us. The v4.2 release notes reference the previous board name and we are changing the name of the board. Is there some way to do an "alias" on the reference?

ZhaoxiangJin commented:
update:
fixed conflict

zejiang0jason commented:
The changes are split into several patches, do we need to merge the necessary patches, to make every snapshot buildable?

ZhaoxiangJin committed:
boards: nxp: Move board specific files from hal to zephyr

Move board specific files (dcd, flash_config, etc.) from
nxp hal repo to zephyr repo.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin commented:
Update:
1. Move board specific files (dcd, flash_config, etc.) from nxp hal repo to zephyr repo.
2. Removed frdm_mcxa166  and frdm_mcxa276 from release-notes-4.2.rst.

dleach02 commented:
@ZhaoxiangJin please address the compliance failures

hakehuang commented:
> @hakehuang can we run one final regression against this PR. If it passes then please apply your +1

testing is in progress, will feedback once done

hakehuang reviewed (approved):
board testing and regression build pass
